### PR TITLE
Select / Combobox refinements

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* clears selection range for firefox bug
* Fixes when we should apply on the onBlur callback on the Combobox input (hint it's not on blur)
* fix some pointer styles

### Why is it needed?

* some bugs @gu-stav found

### CODE REVIEW

**Review by gu-stav — COMMENTED** [review]
Code lgtm.

- [x] the cursor problem exists on the combobox too
- [ ] the reordering problem is still present: https://design-system-git-fix-select-fixes-strapijs.vercel.app/?path=/story/design-system-components-select--multiple

the values are sorted in the order in which they were selected and after closing the popover they become sorted by the order of their appearance

https://user-images.githubusercontent.com/2244375/234582981-920d2432-6fc1-4f39-903c-89a1f190af3c.mp4

File: packages/primitives/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -454,7 +454,7 @@ const SelectValue = React.forwardRef<SelectValueElement, SelectValueProps>(
      */
     return (
       <Primitive.span {...valueProps} ref={composedRefs}>
-        {renderValue ? <span>{renderValue}</span> : null}
+        {renderValue || null}
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/primitives/src/components/Select/Select.tsx:457]: no, it'll be something or `undefined` and then the Portal will render things

**Comment by joshuaellis** [comment]
> the values are sorted in the order in which they were selected and after closing the popover they become sorted by the order of their appearance

This is by design, in the story we're not rendering anything as the child of `Select.Value` so by default it render's the `Select.Item`'s textValue into the `Select.Value`, of course this happens by order or rendering (the list order) IF the order matters you would need to pass a `customiseContent` function. You can see it doesn't do this on the `tags` example :)

**Review by gu-stav — APPROVED** [review]
No 💩 only ❤️

**Pull request merged**